packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -428,9 +493,22 @@ class AuroraPostgresClusterEngine extends ClusterEngineBase {
           majorVersion: version.auroraPostgresMajorVersion,
         }
         : undefined,
+      features: version ? version._features : { s3Import: 's3Import', s3Export: 's3Export' },
     });
   }
 
+  public bindToCluster(scope: core.Construct, options: ClusterEngineBindOptions): ClusterEngineConfig {
+    const config = super.bindToCluster(scope, options);
+    // skip validation for unversioned as it might be supported/unsupported. we cannot reliably tell at compile-time
+    if (options.s3ImportRole && this.engineVersion && !(config.features?.s3Import)) {
```

Comment:
I'd rather not repeat the `this.engineVersion` condition if possible... so:

```suggestion
    if (this.engineVersion?.fullEngineVersion) {
      if (options.s3ImportRole && !(config.features?.s3Import)) {
        throw new Error(`s3Import is not supported for Postgres version: ${this.engineVersion.fullVersion}. Use a version that supports the s3Import feature.`);
      }
      if (options.s3ExportRole && this.engineVersion && !(config.features?.s3Export)) {
        throw new Error(`s3Export is not supported for Postgres version: ${this.engineVersion.fullVersion}. Use a version that supports the s3Export feature.`);
      }
    }